### PR TITLE
Add sane associations for models

### PR DESCRIPTION
**Pull request by DerekStride**
How do these associations look?

### CODE REVIEW

**Review by ismail-syed — APPROVED** [review]

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :content, polymorphic: true
+
+  scope :posts, -> { where(content_type: 'Post') }
+  scope :photos, -> { where(content_type: 'Photo') }
```
Comment (justinkrol) [inline app/models/comment.rb:6]: Let's just remove these, kinda hard to reason about. Talked IRL

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,7 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: :user_id
+  has_many :wall_posts, foreign_key: :receiver_id
```
Comment (justinkrol) [inline app/models/user.rb:3]: How does this association work?

Comment (DerekStride) [reply app/models/user.rb:3]: This is to get all the posts on someones "wall".

Comment (ismail-syed) [reply app/models/user.rb:3]: whats the difference with `posts` and `wall_posts`?

Comment (DerekStride) [reply app/models/user.rb:3]: Posts are posts that the user made
Wall Posts are posts on the users wall (could be renamed to timeline, or just wall)
```ruby
has_many :wall, foreign_key: :receiver_id, class_name: 'Post'
```

**Pull request merged**